include/picongpu/particles/boundary/Absorbing.hpp

```diff
@@ -83,6 +90,167 @@ namespace picongpu
+                 */
+                struct DampWeightsInPml : public functor::misc::Parametrized<DampWeightsInPmlParameters>
+                {
+                    //! Name, required to be wrapped later
+                    static constexpr char const* name = "dampWeightsInPml";
+
+                    /** Process the current particle located in the given cell
+                     *
+                     * @param offsetToTotalOrigin offset of particle cell in the total domain
+                     * @param particle handle of particle to process (can be used to change attribute values)
+                     */
+                    template<typename T_Particle>
+                    DINLINE void operator()(DataSpace<simDim> const& offsetToTotalOrigin, T_Particle& particle)
+                    {
+                        auto const axis = m_parameters.axis;
+                        // Velocity half time step back relative to the current position
+                        auto velocity = Velocity{};
+                        auto const vel
+                            = velocity(particle[momentum_], attribute::getMass(particle[weighting_], particle))[axis];
+
+                        // Positions in local domain, in units of cells, without guards
+                        auto const localCellIdx = offsetToTotalOrigin - m_parameters.localToTotalDomainOffset;
+                        auto const positionCurrent
+                            = (precisionCast<float_X>(localCellIdx) + particle[position_])[axis];
+                        auto const positionPrevious = positionCurrent - vel * DELTA_T / cellSize[axis];
+
+                        // Integral over the last time step as described in comment of this struct
+                        auto const timeStepIntegral = fields::absorber::pml::getNormalizedSigmaIntegral(
+                            positionPrevious,
+                            positionCurrent,
+                            DELTA_T,
+                            m_parameters.localParameters,
+                            axis);
+
+                        // Adjust weighting according to [Lehe2022], cap at MIN_WEIGHTING
+                        auto const weightingMultiplier = math::exp(-timeStepIntegral);
+                        auto const dampedWeighting = particle[weighting_] * weightingMultiplier;
+                        auto const minWeighting = MIN_WEIGHTING; // to avoid taking address of constexpr in max()
+                        /* Update macroparticle weighting and values of all weighted attributes.
+                         * We currently have no generic and consistent way to do that #4299.
+                         * So at least update weighted momentums so that particles don't artificially accelerate.
+                         */
+                        particle[weighting_] = math::max(dampedWeighting, minWeighting);
+                        particle[momentum_] *= weightingMultiplier;
```

Comment:
This modification must take into account that we change the weighting only to a value larger than `minWeighting`.

Reply:
discussed offline @sbastrakov will provide a function in a separate PR to sett he weighing which is then fixing this problem too.